internal/factsengine/gatherers/saphostctrl_test.go

```diff
@@ -0,0 +1,198 @@
+package gatherers_test // nolint:dupl
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	"github.com/trento-project/agent/internal/factsengine/gatherers"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	utilsMocks "github.com/trento-project/agent/pkg/utils/mocks"
+)
+
+type SapHostCtrlTestSuite struct {
+	suite.Suite
+	mockExecutor *utilsMocks.CommandExecutor
+}
+
+func TestSapHostCtrlTestSuite(t *testing.T) {
+	suite.Run(t, new(SapHostCtrlTestSuite))
+}
+
+func (suite *SapHostCtrlTestSuite) SetupTest() {
+	suite.mockExecutor = new(utilsMocks.CommandExecutor)
+}
+
+func (suite *SapHostCtrlTestSuite) TestSapHostCtrlGatherListInstances() {
+	suite.mockExecutor.On("Exec", "/usr/sap/hostctrl/exe/saphostctrl", "-function", "ListInstances").Return(
+		[]byte(" Inst Info : S41 - 41 - s41app - 785, patch 50, changelist 2091708\n"+
```

Comment:
You could maybe replace this by:
```
const output = ` Inst Info : S41 - 41 - s41app - 785, patch 50, changelist 2091708
 Inst Info : S41 - 40 - s41app - 785, patch 50, changelist 2091708
 Inst Info : HS1 - 11 - s41db - 753, patch 819, changelist 2069355`
```
The `+` thing looks ugly

Reply:
I'm not sure I agree with this one, IMHO it looks worse because you can't indent the code properly :/

Reply:
But that's a common thing in golang for this case.
Anyway, up to you. I'm fine with both options